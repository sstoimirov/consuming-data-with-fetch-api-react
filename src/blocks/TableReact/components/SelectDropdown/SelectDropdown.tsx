import * as React from "react";

export class SelectDropdown extends React.PureComponent<{
    onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
    name: string,
    userName: string,
    email: string
}, {}>{

    render() {
        const { name, userName, email } = this.props;
        return (
            <select className="select-options" onChange={this.props.onChange} >
                <option value="name">{name}</option>
                <option value="username">{userName}</option>
                <option value="email">{email}</option>
            </select>
        )
    }
}