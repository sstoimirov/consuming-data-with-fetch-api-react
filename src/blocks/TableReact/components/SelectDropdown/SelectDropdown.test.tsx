import * as React from "react";
import { shallow, configure } from "enzyme";
import { SelectDropdown } from "./SelectDropdown";
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe("SelectDropdown", () => {
    it("Renders correctly", () => {
        const props = {
            onChange: jest.fn(),
            name: "name",
            userName: "username",
            email: "email"
        }
        
        const comp = shallow(<SelectDropdown {...props} />);
        expect(comp.debug()).toMatchSnapshot();
    })
})