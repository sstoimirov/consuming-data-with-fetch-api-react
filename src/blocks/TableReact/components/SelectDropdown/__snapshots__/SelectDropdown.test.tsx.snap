// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDropdown Renders correctly 1`] = `
"<select className=\\"select-options\\" onChange={[Function: mockConstructor]}>
  <option value=\\"name\\">
    name
  </option>
  <option value=\\"username\\">
    username
  </option>
  <option value=\\"email\\">
    email
  </option>
</select>"
`;
